### PR TITLE
chore(bug-report): bug report template yaml

### PR DESCRIPTION
**Pull request by annawen1**
Closes #[4775](https://github.com/carbon-design-system/ibm-products/issues/4775)

Convert bug report to yaml file, including form elements and adding required validation to certain inputs

#### What did you change?

- new `bug-report.yaml` file
- delete `bug-report.md` file

#### How did you test and verify your work?

You can view the form in preview mode on my fork here: https://github.com/annawen1/ibm-products/blob/feat/yaml-bug-template/.github/ISSUE_TEMPLATE/bug-report.yaml

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        blank any sections that aren’t relevant but try to include as much
+        detail as possible. The more details we have, the faster we can respond.'
+  - type: dropdown
+    id: package
+    attributes:
+      label: Package
+      description: Which package(s) are you using?
+      multiple: true
+      options:
+        - 'Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)'
```
[the remark this answers is not in the document]
Comment (elycheea) [reply .github/ISSUE_TEMPLATE/bug-report.yaml:23]: Maybe we can remove it here and keep it in the version section? 👀

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          functionality, has a workaround.'
+        - 'Severity 4 = The problem is not visible to or noticeable to an
+          average user. Affects minor functionality, no workaround needed.'
+    validations:
+      required: true
+  - type: input
+    id: application
+    attributes:
+      label: Application/website
+      description: 'What application/website do you work on?'
```
Comment (elycheea) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:87]:
```suggestion
      label: Product/offering
      description: 'What product/offering do you work on?'
```

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    validations:
+      required: true
+  - type: input
+    id: example-url
+    attributes:
+      label: CodeSandbox example
```
Comment (elycheea) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:93]:
```suggestion
      label: CodeSandbox or Stackblitz example
```

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      label: Code of Conduct
+      description: Please confirm the following
+      options:
+        - label:
+            I agree to follow this project's [Code of
+            Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
+          required: true
+        - label:
+            I checked the [current
+            issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues)
```
Comment (elycheea) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:125]:
```suggestion
            issues](https://github.com/carbon-design-system/ibm-products)
```

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    id: example-url
+    attributes:
+      label: CodeSandbox example
+      description:
+        'Our [Carbon for IBM Products
+        Storybook](https://carbon-for-ibm-products.netlify.app/) documentation
+        includes links to codesandboxes for each component. For example the
+        SidePanel codesandbox
+        [here](https://codesandbox.io/s/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/SidePanel).
+        Use those to help re-create the issue.'
```
Comment (elycheea) [inline .github/ISSUE_TEMPLATE/bug-report.yaml:100]:
```suggestion
        Use those to help re-create the issue or include a link to the
        issue in Storybook.'
```

**Review by elycheea — COMMENTED** [review]
Just a few comments, but one link to dotcom that needs to be updated.

Also, trying to figure out where we can shed the older name (Carbon for IBM Cloud & Cognitive) where we can. 🙈

File: .github/ISSUE_TEMPLATE/bug-report.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (annawen1) [reply .github/ISSUE_TEMPLATE/bug-report.yaml]: yep, good point I'll remove it from the dropdown here and add the language in the version section

**Review by elycheea — APPROVED** [review]

**Pull request merged**